crates/astria-composer/src/searcher/executor.rs

```diff
@@ -120,50 +124,59 @@ impl Executor {
         self.status.subscribe()
     }
 
-    /// Gets the next nonce to sign over if it exists and increments the stored nonce counter
-    fn get_and_increment_nonce(&mut self) -> Option<u32> {
-        self.nonce.map(|curr_nonce| {
+    /// Gets the next nonce to sign over and increments the internal counter.
+    /// If there is not a nonce currently stored, will fetch the latest nonce from the sequencer.
+    async fn get_and_increment_nonce(&mut self) -> eyre::Result<u32> {
+        if let Some(curr_nonce) = self.nonce {
             self.nonce = Some(curr_nonce + 1);
-            curr_nonce
-        })
-    }
-
-    /// Sugmits a signed transaction to the sequencer node.
-    /// TODO: handle failed tx submission due to nonce
-    async fn submit_tx(&self, signed_tx: SignedTransaction) -> eyre::Result<()> {
-        let rsp = self
-            .sequencer_client
-            .submit_transaction_sync(signed_tx)
-            .await
-            .wrap_err("failed submitting transaction to sequencer")?;
-        if rsp.code.is_err() {
-            bail!(
-                "submitting transaction to sequencer returned with error code; code: `{code}`; \
-                 log: `{log}`; hash: `{hash}`",
-                code = rsp.code.value(),
-                log = rsp.log,
-                hash = rsp.hash,
-            );
+            Ok(curr_nonce)
+        } else {
+            let rsp = self.sequencer_client.get_latest_nonce(self.address).await?;
```

Comment:
See my other comment: it's not necessary to wrap this in an `eyre::Result`. Why not just return the error returned by `get_latest_nonce` itself?